process/block/preprocess/rewardTxPreProcessor.go

```diff
@@ -0,0 +1,455 @@
+
+	return nil
+}
+
+// RequestTransactionsForMiniBlock requests missing reward transactions for a certain miniblock
+func (rtp *rewardTxPreprocessor) RequestTransactionsForMiniBlock(mb block.MiniBlock) int {
+	missingRewardTxsForMiniBlock := rtp.computeMissingRewardTxsForMiniBlock(mb)
+	rtp.onRequestRewardTx(mb.SenderShardID, missingRewardTxsForMiniBlock)
+
+	return len(missingRewardTxsForMiniBlock)
+}
+
+// computeMissingRewardTxsForMiniBlock computes missing reward transactions for a certain miniblock
+func (rtp *rewardTxPreprocessor) computeMissingRewardTxsForMiniBlock(mb block.MiniBlock) [][]byte {
+	missingRewardTxs := make([][]byte, 0)
+	if mb.Type != block.RewardsBlockType {
+		return missingRewardTxs
+	}
+
+	for _, txHash := range mb.TxHashes {
+		tx, _ := process.GetTransactionHandlerFromPool(
+			mb.SenderShardID,
+			mb.ReceiverShardID,
+			txHash,
+			rtp.rewardTxPool,
+		)
+
+		if tx == nil {
+			missingRewardTxs = append(missingRewardTxs, txHash)
+		}
+	}
+
+	return missingRewardTxs
+}
+
+// getAllRewardTxsFromMiniBlock gets all the reward transactions from a miniblock into a new structure
+func (rtp *rewardTxPreprocessor) getAllRewardTxsFromMiniBlock(
+	mb *block.MiniBlock,
+	haveTime func() bool,
+) ([]*rewardTx.RewardTx, [][]byte, error) {
+
+	strCache := process.ShardCacherIdentifier(mb.SenderShardID, mb.ReceiverShardID)
+	txCache := rtp.rewardTxPool.ShardDataStore(strCache)
+	if txCache == nil {
+		return nil, nil, process.ErrNilRewardTxDataPool
+	}
+
+	// verify if all reward transactions exists
+	rewardTxs := make([]*rewardTx.RewardTx, 0)
+	txHashes := make([][]byte, 0)
+	for _, txHash := range mb.TxHashes {
+		if !haveTime() {
+			return nil, nil, process.ErrTimeIsOut
+		}
+
+		tmp, _ := txCache.Peek(txHash)
+		if tmp == nil {
+			return nil, nil, process.ErrNilRewardTransaction
+		}
+
+		tx, ok := tmp.(*rewardTx.RewardTx)
+		if !ok {
+			return nil, nil, process.ErrWrongTypeAssertion
+		}
+
+		txHashes = append(txHashes, txHash)
+		rewardTxs = append(rewardTxs, tx)
+	}
+
+	return rewardTxs, txHashes, nil
+}
+
+// CreateAndProcessMiniBlock creates the miniblock from storage and processes the smartContractResults added into the miniblock
+func (rtp *rewardTxPreprocessor) CreateAndProcessMiniBlock(sndShardId, dstShardId uint32, spaceRemained int, haveTime func() bool, round uint64) (*block.MiniBlock, error) {
+	return nil, nil
+}
+
+// ProcessMiniBlock processes all the reward transactions from a miniblock and saves the processed reward transactions
+// in local cache
+func (rtp *rewardTxPreprocessor) ProcessMiniBlock(miniBlock *block.MiniBlock, haveTime func() bool, round uint64) error {
+	if miniBlock.Type != block.RewardsBlockType {
+		return process.ErrWrongTypeInMiniBlock
+	}
+
+	miniBlockRewardTxs, miniBlockTxHashes, err := rtp.getAllRewardTxsFromMiniBlock(miniBlock, haveTime)
+	if err != nil {
+		return err
+	}
+
+	for index := range miniBlockRewardTxs {
+		if !haveTime() {
+			err = process.ErrTimeIsOut
```

Comment:
This could be a one liner `return process.ErrTimeIsOut`

Reply:
done